compiler/rustc_error_codes/src/error_codes/E0787.md

```diff
@@ -0,0 +1,28 @@
+An unsupported naked function definition.
+
+Erroneous code example:
+
+```compile_fail,E0787
+#![feature(naked_functions)]
+
+#[naked]
+pub extern "C" fn f() -> u32 {
+    42
+}
+```
+
+The naked functions must be defined using a single inline assembly
+block.
+
+The execution must never fall through past the end of the assembly
+code so the block must use `noreturn` option. The asm block can also
+use `att_syntax` option, but other options are not allowed.
```

Comment:
The `raw` option is also allowed (or at least should be), which disables format string parsing on the template string.

Reply:
@Amanieu I agree. However, I am trying to understand what precisely we need to accomplish before stabilization. We can stabilize an overly restrictive subset of naked functions and loosen restrictions later where they make sense.

Reply:
I checked the code and `raw` is already allowed in naked functions. It's just the text here that needs to be fixed.

Regarding stabilization I am happy with the way naked functions are right now.

Reply:
@joshtriplett @nikomatsakis FYI ^; do we need to make a stabilization proposal somewhere?

Reply:
Thanks, clarified that `raw` option is also allowed.

Reply:
@npmccallum Yes, a stabilization report as part of the PR to stabilize. That will also need some indications of active use in nightly.